tests/test_code_references.py

```diff
@@ -203,114 +208,123 @@ def x(my_class):
 
     assert get_references(x) == ["val", logging.info, logging.debug]
 
+    import math
+
+    def y(op_type, val):
+        if op_type == "ceil":
+            func = math.ceil
+        else:
+            func = math.floor
+
+        return func(val)
+
+    # TODO: We should also return math.ceil as a reference. It isn't
+    # returned now because it is stored in a local variable. Maybe we
+    # should return the values of local variables as they are assigned.
+    assert get_references(y) == [math.floor]
+
 
 def test_complex_function():
-    import inspect
-    import warnings
-    from bionic.code_hasher import CodeHasher, TypePrefix
-    from bionic.utils.reload import is_internal_file
-
-    hasher = CodeHasher(False)
-    # TODO: This makes me realize that maybe we should dedup the results.
-    assert get_references(hasher._ingest) == [
-        "isinstance",
-        "bytes",
-        hasher._ingest_raw_prefix_and_bytes,
-        TypePrefix.BYTES,
-        "isinstance",
-        "bytearray",
-        hasher._ingest_raw_prefix_and_bytes,
-        TypePrefix.BYTEARRAY,
-        hasher._ingest_raw_prefix_and_bytes,
-        TypePrefix.NONE,
-        "isinstance",
+    # Example code of a ml flow. Taken from below article with minor changes:
```

Comment:
I like the realism of this example, but one weakness it has is that there's no control flow. For example, it wouldn't surface the `math.ceil` issue you found above because it doesn't have any if statements -- right?

Reply:
Right. I will add a control flow to find the prediction model.